### DISCUSSION TITLE
Customize repeated half action cycles

### DISCUSSION THREAD

**Discussion by alexisgaziello** [rxhanson/Rectangle#532] [Ideas]
Hey!

I am a big fan of this app. One of the features I use the most is the "repeated half action cycles". But I would like to customize those cycles a little. Instead of rotating over 3 positions, I would like to expand it to 4, being the last one a window that would cover the screen horizontally around 7/8s.

Of course, I am not assuming that not everybody will want this fourth option, so the idea would be to add an option to customize the steps and fractions of the cycle.

I have seen that the cycle is defined in the protocol RepeatedExecutionsInThirdsCalculation. The idea would be to get the parameters of this protocol dynamically instead of hard-coding them.

If I work on a feature to customize the cycle through some kind of table that would appear when pressing a button in the preferences, would it be accepted?

Thanks,
Alexis

**Comment by rxhanson** [comment]
Thanks for looking to contribute to Rectangle!

This type of behavior is easily achieved in the custom shortcut builder in my paid superset of Rectangle, [Hookshot](https://hookshot.app). 

If you're really wanting to add this in, I would consider a pull request, but I wouldn't want to add any changes to the user interface. At first cut this would mean something configurable only using terminal commands.

**Reply by alexisgaziello** [reply]
Awesome! I will look into it.
We could have something like:

`default write com.knollsoft.Rectangle cycle -string "1/2 2/3 1/3"`

`default write com.knollsoft.Rectangle cycle -string "1/2 2/3 7/8 1/3"`

And then maybe parse that string in the app. What do you think? I have no idea if this is possible (or the best way to do it). 

Maybe an array would be more appropriate?

`default write com.knollsoft.Rectangle cycle -array [0.5, 0.66, 0.85, 0.33]`

I have worked with software but not with OS X apps. I would greatly appreciate it if you could point me in the right direction. Maybe files I should check, links that I could use as a reference.

Thank you!

**Reply by alexisgaziello** [reply]
Or maybe I should just get Hookshot 😂
